### DISCUSSION TITLE
BasicInput 의 error 여부를 외부에서 주입하지 못하는 이슈가 있습니다

### DISCUSSION THREAD

**Discussion by kutta97** [prgrms-fe-devcourse/FEDC4_CUCUMIS_Pocojang#57] [General]
### 문제 상황
SignupPage 개발 중, BasicInput 을 사용하여 password 와 password confirm 을 개발 중
password confirm 의 error 여부를 바깥에서 계산하여 주입할 수 없었습니다.

```typescript
<BasicInput
  type="password"
  label="password"
  regExp="^(?=.*[a-zA-Z])(?=.*[0-9]).{8,25}$"
  errorMessage="영문, 숫자 포함 몇 자 이상 validation"
  isRequired
/>
<BasicInput
  type="password confirm"
  label="password confirm"
  regExp=""
  errorMessage="영문, 숫자 포함 몇 자 이상 validation"
  isRequired
/>
```

### 문제 해결 방법 제안
`useForm` hook 을 사용해서 하나의 hook 으로 여러 input 값들과 input 값들의 에러 여부를 관리할 수 있도록 개선할 수 있으면 좋을 것 같습니다!

### BaseInput 관련 추가 이슈
BaseInput 컴포넌트에 placeholder attribute 가 없는 것 같습니다ㅠㅠ
BaseInput 컴포넌트의 prop 이 TextFieldProps 를 extends 하고 있지 않은 것으로 보입니다...

@1g2g

**Comment by 1g2g** [comment]
해결했습니다! 훅 만들 때는 사용성과 확장성을 더 고려해서 만들어보겠습니다~
